ydb/library/yql/yt/native/dq_manager.cpp

```diff
@@ -0,0 +1,261 @@
+#include "dq_manager.h"
+
+#include <ydb/library/yql/providers/dq/actors/yt/resource_manager.h>
+#include <ydb/library/yql/providers/dq/actors/dynamic_nameserver.h>
+#include <ydb/library/yql/providers/dq/metrics/metrics_printer.h>
+#include <ydb/library/yql/providers/dq/provider/yql_dq_control.h>
+#include <ydb/library/yql/providers/dq/service/interconnect_helpers.h>
+#include <ydb/library/yql/providers/dq/stats_collector/pool_stats_collector.h>
+#include <ydb/library/yql/providers/dq/worker_manager/interface/events.h>
+
+#include <ydb/library/yql/providers/yt/dq_task_preprocessor/yql_yt_dq_task_preprocessor.h>
+
+#include <ydb/library/yql/minikql/invoke_builtins/mkql_builtins.h>
+
+#include <ydb/library/yql/utils/backtrace/backtrace.h>
+#include <ydb/library/yql/utils/log/log.h>
+#include <ydb/library/yql/utils/log/proto/logger_config.pb.h>
+
+#include <library/cpp/digest/md5/md5.h>
+
+namespace NYT::NYqlPlugin {
+
+using namespace NYql;
+using namespace NYql::NDqs;
+
+//////////////////////////////////////////////////////////////////////////////
+
+void TDqManagerConfig::Register(TRegistrar registrar)
+{
+    registrar.Parameter("interconnect_port", &TThis::InterconnectPort)
+        .GreaterThan(0);
+    registrar.Parameter("grpc_port", &TThis::GrpcPort)
+        .GreaterThan(0);
+    registrar.Parameter("actor_threads", &TThis::ActorThreads)
+        .GreaterThan(0);
+    registrar.Parameter("use_ipv4", &TThis::UseIPv4)
+        .Default(false);
+
+    registrar.Parameter("yt_backends", &TThis::YtBackends)
+        .NonEmpty();
+
+    registrar.Parameter("yt_coordinator", &TThis::YtCoordinator)
+        .Default();
+    registrar.Parameter("scheduler", &TThis::Scheduler)
+        .Default();
+    registrar.Parameter("interconnect_settings", &TThis::ICSettings)
+        .Default();
+}
+
+//////////////////////////////////////////////////////////////////////////////
+
+TDqManager::TDqManager(const TDqManagerConfigPtr& config)
```

Comment:
Having such hard logic in constructor is a somewhat discouraged practice. Let's introduce another method "void Start()". Constructor must have "static" initialization and, optionally, parsing of configs, while Start() must actually have some externally visible effect, e.g. perform some network activity, expose this coordinator or start vanilla operations.